instrumentation/armeria-0.99/library/src/main/java/io/opentelemetry/instrumentation/armeria/v0_99/server/OpenTelemetryService.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.armeria.v0_99.server;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.logging.RequestLogProperty;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Route;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingHttpService;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+/** Decorates an {@link HttpService} to trace inbound {@link HttpRequest}s. */
+public class OpenTelemetryService extends SimpleDecoratingHttpService {
+
+  /** Creates a new tracing {@link HttpService} decorator using the default {@link Tracer}. */
+  public static Function<? super HttpService, OpenTelemetryService> newDecorator() {
+    return service -> new OpenTelemetryService(service, new ArmeriaServerTracer());
+  }
+
+  /** Creates a new tracing {@link HttpService} decorator using the specified {@link Tracer}. */
+  public static Function<? super HttpService, OpenTelemetryService> newDecorator(Tracer tracer) {
+    return service -> new OpenTelemetryService(service, new ArmeriaServerTracer(tracer));
+  }
+
+  private final ArmeriaServerTracer serverTracer;
+
+  private OpenTelemetryService(HttpService delegate, ArmeriaServerTracer serverTracer) {
+    super(delegate);
+    this.serverTracer = serverTracer;
+  }
+
+  @Override
+  public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+    String route = route(ctx);
+    String spanName = route != null ? route : "HTTP " + req.method().name();
```

Comment:
Curious if we can utilize `ctx.log().serviceName()`

Reply:
Yeah let me check it in a followup PR where I add tests for armeria-grpc with it, for now I want to focus on the minimum HTTP behavior. Especially with auto instrumentation we might have some issue due to applying both instrumentations.

In the meantime maybe you can help with this one :)
 https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/756/files#diff-a63d5fb32f3eb41b3cde769b7807b2e6R87